core/comlink/src/typescript_parser/parser.rs

```diff
@@ -0,0 +1,1188 @@
+        let mut output = JsonValue::Null;
+        let mut spans = UseCaseExampleSpans::default();
+        spans.entire = root.range().into();
+
+        for member in root.members() {
+            let member = ast_do!(UseCaseExampleInvalid, member; [
+                mtch(Ok, root),
+                mtch(AnyJsObjectMember::JsPropertyObjectMember),
+            ] catch(err) {
+                self.diag.error(err);
+                return None;
+            });
+
+            let member_name = ast_do!(UseCaseExampleInvalid, &member; [
+                err(.name),
+                or(.name),
+            ] catch(err) {
+                self.diag.error(err);
+                return None;
+            });
+
+            let member_value = ast_do!(UseCaseExampleInvalid, &member; [
+                err(.value)
+            ] catch(err) {
+                self.diag.error(err);
+                return None;
+            });
+
+            match member_name.deref() {
+                "input" => {
+                    spans.input = member_value.range().into();
+                    input = self.parse_example_element_value(member_value);
+                }
+                "result" => {
+                    is_error = false;
+                    spans.output = member_value.range().into();
+                    output = self.parse_example_element_value(member_value);
+                }
+                "error" => {
+                    is_error = true;
+                    spans.output = member_value.range().into();
+                    output = self.parse_example_element_value(member_value);
+                }
+                "name" => match ast_do!(UseCaseExampleInvalid, member_value; [
+                    mtch(AnyJsExpression::AnyJsLiteralExpression),
+                    mtch(AnyJsLiteralExpression::JsStringLiteralExpression),
+                    err(.value_token),
+                ]) {
+                    Ok(v) => {
+                        name = Some(inner_string_text(&v).text().into());
+                        spans.name = v.text_trimmed_range().into();
+                    }
+                    Err(err) => {
+                        self.diag
+                            .error_detail(err, "example name must be a string literal");
+                    }
+                },
+                _ => self.diag.warn(PartialDiagnostic(
+                    DiagnosticCode::UseCaseExampleMemberUnknown,
+                    member.range(),
+                )),
+            }
+        }
+
+        let example = if is_error {
+            UseCaseExample::Failure {
+                name,
+                input,
+                error: output,
+            }
+        } else {
+            UseCaseExample::Success {
+                name,
+                input,
+                result: output,
+            }
+        };
+
+        Some((example, spans))
+    }
+
+    fn parse_example_element_value(&mut self, expr: AnyJsExpression) -> JsonValue {
```

Comment:
I don't even remember we allowed to have expressions in example values.

Reply:
Even literals are wrapped in expression node when in expression position. Since examples are of the form:
```ts
const examples: UseCaseName['examples'] = [
  { input: 1, result: 2 }
]
```
Then all of `[]`, `{}`, `1` and `2` are something like `Expression(Literal)` nodes in the tree.